.github/workflows/tests.yml

```diff
@@ -64,8 +64,14 @@ jobs:
         if: steps.cache.outputs.cache-hit == 'true' && matrix.os != 'MacOS'
         run: timeout 10s poetry run pip --version || rm -rf .venv
 
-      - name: Install dependencies
-        run: poetry install --only main --only test -vvv
+      - name: Install typing dependencies
+        run: poetry install --only main --only test --only typing -vvv
+
+      - name: Run type checking
+        run: poetry run mypy
+
+      - name: Install runtime and testing dependencies
+        run: poetry install --only main --only test --sync -vvv
```

Comment:
Not sure if we need to install project twice. We waste CI time for something that doesn't give us that much benefit

Reply:
This line could be changed to `poetry install --only main --only test --sync --no-root -vvv` so that the build only occurs once